packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -0,0 +1,855 @@
+    Size? preferredSize,
+    BoxConstraints? preferredConstraints,
+  }) {
+    throw UnsupportedError(errorMessaage);
+  }
+
+  @override
+  bool hasTopLevelWindows() {
+    throw UnsupportedError(errorMessaage);
+  }
+}
+
+/// The [RegularWindow] widget provides a way to render a regular window in the
+/// widget tree.
+///
+/// The provided [controller] creates the native window that backs
+/// the widget. The [child] widget is rendered into this newly created window.
+///
+/// When a [RegularWindow] widget is removed from the tree, the window that was created
+/// by the [controller] remains valid until the caller destroys it by calling
+/// [RegularWindowController.destroy].
+///
+/// Widgets in the same tree as the [child] widget will have access to the
+/// [RegularWindowController] via the [WindowControllerScope] widget.
+///
+/// An example usage might look like:
+/// ```dart
+/// final RegularWindowController controller = RegularWindowController(
+///   preferredSize: Size(800, 600),
+///   preferredConstraints: BoxConstraints(minWidth: 640, minHeight: 480),
+///   title: "Example Window",
+/// );
+/// runWidget(
+///   RegularWindow(
+///     controller: controller,
+///     child: MaterialApp(home: Container()),
+///   ),
+/// );
+/// ```
+///
+/// {@macro flutter.widgets.windowing.experimental}
+@internal
+class RegularWindow extends StatelessWidget {
+  /// Creates a regular window widget.
+  ///
+  /// The [controller] creates the native backing window into which the
+  /// [child] widget is rendered.
+  ///
+  /// It is up to the caller to destroy the window by calling
+  /// [RegularWindowController.destroy] when the window is no longer needed.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  RegularWindow({super.key, required this.controller, required this.child}) {
+    if (!isWindowingEnabled) {
+      throw UnsupportedError(_kWindowingDisabledErrorMessage);
+    }
+  }
+
+  /// Controller for this widget.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final RegularWindowController controller;
+
+  /// The content rendered into this window.
+  ///
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  final Widget child;
+
+  /// {@macro flutter.widgets.windowing.experimental}
+  @internal
+  @override
+  Widget build(BuildContext context) {
+    return View(
+      view: controller.rootView,
+      child: WindowControllerScope(controller: controller, child: child),
+    );
+  }
+}
+
+enum _WindowControllerAspect { contentSize, title, activated, maximized, minimized, fullscreen }
+
+/// Provides descendants with access to the [BaseWindowController] associated with
+/// the window that is being rendered.
+///
+/// {@macro flutter.widgets.windowing.experimental}
+///
+/// See also:
+///
+///  * [RegularWindow], the widget to create a regular window.
+@internal
+class WindowControllerScope extends InheritedModel<_WindowControllerAspect> {
```

Comment:
What do we think of the name `Window` here instead of `WindowControllerScope`?

I know that today we only store the `BaseWindowController` in this inherited model, but in the future we might decide to store more information too.

Switching to the name `Window` would let you write `Window.isFullscreenOf(context)`, which seems significantly nicer than `WindowControllerScope.isFullScreenOf(context)`.

If you rename this type, I would rename `of`/`maybeOf` to `controllerOf`/`controllerMaybeOf`.